From 2bb4de710f2b933654951f76adff680d14e62c25 Mon Sep 17 00:00:00 2001
From: Ahajha <ahajha@gmail.com>
Date: Sun, 19 May 2024 03:34:54 -0400
Subject: [PATCH] Add Conan specific CMake changes

Expat needs to use module mode, since at time of writing expat's CMake uses `expat` for find mode and `EXPAT` for module mode. Git looks for the latter.
PkgConfig is only used within CMake for one dependency, so it's easier to just change it to use find_package than deal with PkgConfigDeps.
---
 contrib/buildsystems/CMakeLists.txt | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/contrib/buildsystems/CMakeLists.txt b/contrib/buildsystems/CMakeLists.txt
index 804629c525..d499a1d1a3 100644
--- a/contrib/buildsystems/CMakeLists.txt
+++ b/contrib/buildsystems/CMakeLists.txt
@@ -151,7 +151,7 @@ include(CTest)
 
 find_package(ZLIB REQUIRED)
 find_package(CURL)
-find_package(EXPAT)
+find_package(EXPAT MODULE)
 find_package(Iconv)
 
 #Don't use libintl on Windows Visual Studio and Clang builds
@@ -159,12 +159,9 @@ if(NOT (WIN32 AND (CMAKE_C_COMPILER_ID STREQUAL "MSVC" OR CMAKE_C_COMPILER_ID ST
 	find_package(Intl)
 endif()
 
-find_package(PkgConfig)
-if(PkgConfig_FOUND)
-	pkg_check_modules(PCRE2 libpcre2-8)
-	if(PCRE2_FOUND)
-		add_compile_definitions(USE_LIBPCRE2)
-	endif()
+find_package(PCRE2)
+if(PCRE2_FOUND)
+    add_compile_definitions(USE_LIBPCRE2)
 endif()
 
 if(NOT Intl_FOUND)
-- 
2.34.1

